src/lumigo_tracer/lambda_tracer/lambda_reporter.py

```diff
@@ -275,17 +289,77 @@ def _get_prioritized_spans(
             span_metadata_size = spans_to_send_sizes[index]
             span_size = get_event_base64_size(span)
 
-            if current_size + span_size - span_metadata_size < request_max_size:
+            if current_size + span_size - span_metadata_size < buffered_max_size:
                 spans_to_send_dict[index] = span
                 current_size += span_size - span_metadata_size
             else:
                 # This is an optimization step. If the spans are too big, don't try to send them.
                 too_big_spans += 1
                 if too_big_spans >= too_big_spans_threshold:
                     break
+
+            # If we dropped spans we need to update the enrichment spans dropped spans reasons
+            if len(spans_to_send_dict) != len(msgs):
+                with lumigo_safe_execute(
+                    "create_request_body: smart span selection: updating enrichment span"
+                ):
+                    spans_to_send_dict = _update_enrichment_span_about_prioritized_spans(
+                        spans_to_send_dict, msgs, current_size, buffered_max_size
+                    )
+
     return list(spans_to_send_dict.values())
 
 
+def _update_enrichment_span_about_prioritized_spans(
+    spans_dict: Dict[int, Dict[str, Any]],
+    msgs: List[Dict[str, Any]],
+    current_size: int,
+    max_size: int,
+) -> Dict[int, Dict[str, Any]]:
+    """
+    Looks at the given spans about to be sent + the total number of messages,
+    and updates the enrichment spans about any dropped spans
+    @param spans_dict: The mapping of spans about to be sent
+    @param msgs: The complete list of spans created
+    @param current_size: The current size of all the spans in spans_dict
+    @param max_size: The maximum size of all spans together
+    @return: A modified version of spans_dict input, including the updated enrichment span
+    """
+    spans_to_send_dict = {index: dict(span) for index, span in spans_dict.items()}
```

Comment:
I think this is a waste of time to convert all spans and iterate them twice, mostly that the enrichment span should be the second spans, its better to just do a for loop that run the logic when it see the first enrichment span and add the data to it.

Reply:
Fixed in the refactor of the func